### PR TITLE
Add redux-form declaration

### PR DESCRIPTION
**Pull request by bassettsj**
- Add a basic redux-form declaration to the library with a basic test

### CODE REVIEW

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    viisted: boolean,
+    autofiller: boolean
+  };
+  declare type FormProps = {
+    active: string,
+    asyncValidate: Function,
+    asyncValidating: string | boolean,
+    destroyForm: Function,
+    dirty: boolean,
+    error: String,
```
Comment (jeffmo) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: lower-case-`s`?

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    dirty: boolean,
+    pristine: boolean,
+    active: boolean,
+    touched: boolean,
+    viisted: boolean,
+    autofiller: boolean
+  };
+  declare type FormProps = {
+    active: string,
+    asyncValidate: Function,
```
Comment (jeffmo) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: Per the docs, it looks like this type is:

`(values: Object, dispatch: Function, props: Object) => Promise<void>`. 

Maybe we can be more specific?

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    asyncValidating: string | boolean,
+    destroyForm: Function,
+    dirty: boolean,
+    error: String,
+    fields: { [fieldName: string]: InputProps },
+    handleSubmit: Function,
```
Comment (jeffmo) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: `handleSubmit` takes no params and returns `void`, am I reading that right? If so, could we be specific and type this as `() => void`?

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  };
+  declare type FormConfig = {
+    fields: Array<string>,
+    form: string,
+    alwaysAsyncValidate?: boolean,
+    asyncBlurFields?: Array<string>,
+    asyncValidate?: (values: Object, dispatch: Function, props: Object) => Promise<void>,
+    destroyOnUnmount?: boolean,
+    formKey?: string,
+    getFormState?: Function,
```
Comment (jeffmo) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: This function always takes a state and a mountPoint, right?

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    initialValues?: Object,
```
Comment (jeffmo) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: Docs suggest this is an object with only string values. Could we type this as `{[field: string]: string}`?

**Comment by jeffmo** [comment]
Woohoo! This is awesome

I commented on a few of lines, but in general we try to discourage use of `Function` and `Object` when it's possible to be more specific. The reason is because `Object` can unexpectedly introduce `any`s (i.e. if you access a property on a thing typed as `Object`, the property will have type `any` -- which can flow through the consumer's program and lose type errors). Similarly with `Function`: If you call it, the return type is `any` -- which can flow (allowing type errors) as well.

Anyway, if there are places you want to be more specific that would be sweet. If not, no worries -- we can always merge and iterate later :)

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js (hunk 1 above)
Comment (bassettsj) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: Updated, thank you.

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js (hunk 2 above)
Comment (bassettsj) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: I think that would be a good start at least.

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js (hunk 3 above)
Comment (bassettsj) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: I think it does take params, basically the field names and then their values, then you can return the a promise or void, I believe: http://redux-form.com/5.2.3/#/api/props?_k=omut2m

File: definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js (hunk 4 above)
Comment (bassettsj) [inline definitions/redux-form_v5.x.x/flow_>=v0.22.1/redux-form_v5.x.x.js]: Updated: thanks

**Comment by bassettsj** [comment]
@jeffmo Thanks for the review, I tried updating the definitions to the best of my ability. It should be a bit better.

**Comment by jeffmo** [comment]
thank *you*! Looks great

**Pull request merged**